### PR TITLE
[READY] Cool & good new arcade game

### PR DESCRIPTION
**Pull request by PKPenguin321**
:cl:
add: Keep your eye out for a rare new arcade game!
/:cl:

https://tgstation13.org/phpBB/viewtopic.php?f=9&t=32&view=unread#p484730

Has a 2/100 chance of spawning instead of a regular arcade game

### CODE REVIEW

File: code/game/machinery/computer/arcade.dm [hunk 1]
```diff
@@ -1112,6 +1124,34 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
+/obj/machinery/computer/arcade/amputation
+	name = "Mediborg's Amputation Adventure"
+	desc = "A picture of a blood-soaked medical cyborg flashes on the screen. The mediborg has a speech bubble that says, \"Put your hand in the machine if you aren't a <b>coward!</b>\""
+	icon_state = "arcade"
+	circuit = /obj/item/circuitboard/computer/arcade/amputation
+
+/obj/machinery/computer/arcade/amputation/attack_hand(mob/user)
+	if(!iscarbon(user))
+		return
+	var/mob/living/carbon/c_user
```
Comment (PKPenguin321) [inline code/game/machinery/computer/arcade.dm]: = user

**Comment by Shadowflame909** [comment]
Both of the arcade machines just make bombs as a prize once emagged.


Why not have this replace one of them?

Preferably the Outbomb Cuban Pete variant, because that ones less creative

**Comment by Farquaar** [comment]
Any screenshots of the sprite? Or will it look the same?

**Comment by SpadesNeil** [comment]
image

From the forum.

**Comment by wesoda25** [comment]
Then we can add the one armed bandit.

**Comment by anconfuzedrock** [comment]
part of me wants this to be an emag function since outbomb is abused so easily

**Comment by Dorsisdwarf** [comment]
How does this interact with people who have more than 2 hands?

**Comment by SpadesNeil** [comment]
>How does this interact with people who have more than 2 hands?

You get more than two tries.

I think.

Idk, I didn't code it.

**Comment by PKPenguin321** [comment]
>How does this interact with people who have more than 2 hands?

Probably ignores all the extra hands

File: code/game/machinery/computer/arcade.dm [hunk 2]
```diff
@@ -65,7 +65,19 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
 	// If it's a generic arcade machine, pick a random arcade
 	// circuit board for it and make the new machine
 	if(!circuit)
-		var/choice = pick(subtypesof(/obj/item/circuitboard/computer/arcade))
+		var/list/normalgames = list()
+		var/list/raregames = list()
+		for(var/games in subtypesof(/obj/item/circuitboard/computer/arcade))
+			var/obj/item/circuitboard/computer/arcade/thegame = games//you just lost
```
Comment (Qustinnus) [inline code/game/machinery/computer/arcade.dm]: die

File: code/game/machinery/computer/arcade.dm [hunk 3]
```diff
@@ -65,7 +65,19 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
+			if(thegame.raregame)
+				raregames |= thegame
+			else
+				normalgames |= thegame
+		var/choice
+		if(prob(1))
+			choice = pick(raregames)
+		else
+			choice = pick(normalgames)
```
Comment (Qustinnus) [inline code/game/machinery/computer/arcade.dm]: just make a weighted light and use pickweight, this is gay

Comment (PKPenguin321) [reply code/game/machinery/computer/arcade.dm]: Oh, good idea

File: code/game/machinery/computer/arcade.dm (hunk 2 above)
Comment (PKPenguin321) [reply code/game/machinery/computer/arcade.dm]: Owned

File: code/game/machinery/computer/arcade.dm [hunk 4]
```diff
@@ -1112,6 +1124,34 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
+	if(!iscarbon(user))
+		return
+	var/mob/living/carbon/c_user
+	var/obj/item/bodypart/l_arm = c_user.get_bodypart(BODY_ZONE_L_ARM)
+	var/obj/item/bodypart/r_arm = c_user.get_bodypart(BODY_ZONE_R_ARM)
+	to_chat(c_user, "<span class='warning'>You move your hand towards the machine, and begin to hesitate as a bloodied guillotine emerges from inside of it...</span>")
+	if(do_after(c_user, 50, target = src))
+		to_chat(c_user, "<span class='userdanger'>The guillotine drops on your arm, and the machine sucks it in!</span>")
+		if(l_arm)
+			l_arm.dismember()
```
[the remark this answers is not in the document]
Comment (PKPenguin321) [reply code/game/machinery/computer/arcade.dm]: was too lazy to dig into dismemberment procs and make that happen. is there a good way to get the actual limb from the active hand?

File: code/game/machinery/computer/arcade.dm [hunk 5]
```diff
@@ -1112,6 +1115,36 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
+	var/obj/item/bodypart/r_arm = c_user.get_bodypart(BODY_ZONE_R_ARM)
+	to_chat(c_user, "<span class='warning'>You move your hand towards the machine, and begin to hesitate as a bloodied guillotine emerges from inside of it...</span>")
+	if(do_after(c_user, 50, target = src))
+		to_chat(c_user, "<span class='userdanger'>The guillotine drops on your arm, and the machine sucks it in!</span>")
+		playsound(loc, 'sound/weapons/slice.ogg', 25, 1, -1)
+		if(l_arm)
+			l_arm.dismember()
+			qdel(l_arm)
+		else
+			r_arm.dismember()
```
Comment (deathride58) [inline code/game/machinery/computer/arcade.dm]: This will cause a runtime if the user lacks a right arm

Comment (PKPenguin321) [reply code/game/machinery/computer/arcade.dm]: Good point. Would prefer to fix it by making this whole thing just target the active arm, but again, there doesn't seem to be a way to do that...

File: code/game/machinery/computer/arcade.dm (hunk 4 above)
[the remark this answers is not in the document]
Comment (tralezab) [reply code/game/machinery/computer/arcade.dm]: Active hand index is always odd for left hands, and always even for right. You can grab the limb depending on whether it is even or odd. if you wanted to get funky the indexes come in sets of 2 but i wouldn't even bother, nobody has used extra hands in the code (outside of admin fun) for the ~3 years it has been in

[the remark this answers is not in the document]
Comment (PKPenguin321) [reply code/game/machinery/computer/arcade.dm]: nice, got it working

**Comment by PKPenguin321** [comment]
ready to go

**Review by SpaceManiac — APPROVED** [review]
Tested, interacts sanely with >2 hands

**Pull request merged**